lib/hexpm/web/views/api/retirement_view.ex

```diff
@@ -4,11 +4,21 @@ defmodule Hexpm.Web.API.RetirementView do
   def render("show." <> _, %{retirement: retirement}) do
     render_one(retirement, __MODULE__, "show")
   end
+  def render("minimal." <> _, %{retirement: retirement}) do
+    render_one(retirement, __MODULE__, "minimal")
+  end
 
   def render("show", %{retirement: retirement}) do
     %{
       message: retirement.message,
       reason: retirement.reason,
     }
   end
+
+  def render("minimal", %{retirement: %{retirement: retirement}}) when is_nil(retirement), do: %{}
+  def render("minimal", %{retirement: %{retirement: retirement, version: version}}) do
```

Comment:
I think there's a couple minor issues with this implementation:

1. it's called minimal, which in other views means less data. In this particular case it has the same data as `show.json` plus version so I think it might be a bit confusing in the future.
2. `%{retirement: %{retirement: retirement` looks odd, it should really be: `%{release: %{retirement: retirement`

how about:

```elixir
def render("minimal", %{retirement: %{retirement: retirement, version: version}}) do
```
Alternatively we could have:

```elixir
defmodule RetirementView do
  def render("releases." <> _, %{releases: releases}) do
    releases
    |> Enum.filter(& &1.retirement)
    |> Map.new(&{&1.version, render_one(&1.retirement, __MODULE__, "show")})
  end

  # OR:

  def render("package.", %{package: package}) do
    package.releases
    |> ...
  end
```

but either approach won't work out of the box with `include_if_loaded` helper function.

@ericmj other ideas how to best implement it?

Reply:
I pushed up a couple changes based on your feedback. I will wait to see what Eric says for implementation recommendations.